File: DS-Algorithms/Searching/binary_search/code.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>

int binary_search(int arr[], int arr_size, int element){
    int low = 0;
    int high = arr_size-1;
    int mid;
    while(low <= high){
        mid = low + (high-low)/2;
        if(arr[mid] == element)return mid;
        if(arr[mid] > element) high = mid - 1;
        if(arr[mid] < element) low = mid + 1; 
    }
return -1;
}

int main(){
    srand(time(0));
    int arr_size = (rand()%50000 + 50); //Means size lie between 50 to 149
    int i;
    int *arr = (int*)malloc(sizeof(int)*arr_size);
    int lowerLimit = 10;
    int maxLimit = 50;
    for(i = 0; i < arr_size; ++i){
        arr[i] = (rand()%maxLimit)+lowerLimit;
        lowerLimit = arr[i];
        
    }
    printf("%d" ,binary_search(arr, arr_size, arr[5]));
    printf("\n");
    return 0;
}

File: DS-Algorithms/Searching/binary_search/code.py
import random

def binary_search(arr, element):
    low = 0
    high = len(arr)
    while low <= high:
        mid = low + (high - low)/2
        if arr[mid] == element:
            return mid
        if arr[mid] > element:
            high = mid - 1
        if arr[mid] < element:
            low = mid + 1
    return -1
arr_length = random.randint(50,5000)
arr = []
i = 0
lowerLimit = 10
maxLimit = 50
while i < arr_length:
    arr.append(random.randint(lowerLimit,maxLimit))
    lowerLimit = arr[i]
    maxLimit = lowerLimit + 50
    i = i + 1
print(binary_search(arr,arr[45]))